cmd/tailscale/cli/serve_test.go

```diff
@@ -631,23 +631,59 @@ func TestServeConfigMutations(t *testing.T) {
 			continue
 		}
 		if st.wantErr != nil {
-			t.Fatalf("step #%d, line %v: got success (saved=%v), but wanted an error", i, st.line, newState != nil)
+			t.Fatalf("step #%d, line %v: got success (saved=%v), but wanted an error", i, st.line, lc.config != nil)
 		}
-		if !reflect.DeepEqual(newState, st.want) {
+		if !reflect.DeepEqual(lc.config, st.want) {
 			t.Fatalf("[%d] %v: bad state. got:\n%s\n\nwant:\n%s\n",
-				i, st.command, asJSON(newState), asJSON(st.want))
+				i, st.command, asJSON(lc.config), asJSON(st.want))
 			// NOTE: asJSON will omit empty fields, which might make
 			// result in bad state got/want diffs being the same, even
 			// though the actual state is different. Use below to debug:
 			// t.Fatalf("[%d] %v: bad state. got:\n%+v\n\nwant:\n%+v\n",
-			// i, st.command, newState, st.want)
-		}
-		if newState != nil {
-			current = newState
+			// i, st.command, lc.config, st.want)
 		}
 	}
 }
 
+// fakeLocalServeClient is a fake implementation of LocalClient
+// used by the serve command tests. It's not a full implementation,
+// just enough to test the serve command.
+//
+// The fake client is stateful, and is used to test manipulating
+// ServeConfig state.
+type fakeLocalServeClient struct {
+	mu     sync.Mutex
```

Comment:
why the mutex? If want to keep it, you should fix all the callers above that access the config field without holding it, for consistency.

But I think it's safe to just drop the mutex and document that this implementation can't be used concurrently.

Reply:
Done.